lib/Model/Binary.cpp

```diff
@@ -155,6 +155,21 @@ bool Binary::verifyTypes(VerifyHelper &VH) const {
     auto Name = Type->name();
     if (not Names.insert(Name).second)
       return VH.fail(Twine("Multiple types with the following name: ") + Name);
+
+    switch (Type->Kind()) {
+    case model::TypeKind::CABIFunctionType: {
+      const auto &T = llvm::cast<model::CABIFunctionType>(*Type);
+      if (getArchitecture(T.ABI()) != Architecture())
+        return VH.fail();
+    } break;
+    case model::TypeKind::RawFunctionType: {
+      const auto &T = llvm::cast<model::RawFunctionType>(*Type);
+      if (T.Architecture() != Architecture())
+        return VH.fail();
+    } break;
+    default:
+      break;
+    }
```

Comment:
1. Very common pattern is to as follows:
    ```c++
   if (const auto *T = llvm::dyn_cast<model::CABIFunctionType>(Type)) {
     // ...
   } else if (const auto *T = llvm::dyn_cast<model::CABIFunctionType>(Type)) {
     // ...
   }
   ```
2. Add a failure message in `VH.fail()` these will eventually pop up in the UI  as an explanation (+ swiggly underline) why the model is invalid. Same for all other new validation failure modes that have been introduced.

Reply:
Done.